### PR TITLE
omnibus-f4sd osd integration

### PR DESCRIPTION
**Pull request by DanielePettenuzzo**
This PR adds the driver for the osd chip on the omnibus-f4sd board (MAX7456 or AT7456). The chip is controlled via SPI from the F4 micro controller and adds an overlay to the analog FPV video stream before it is transmitted from the transmitter on the vehicle.

Currently the battery voltage, the accumulated current, altitude and flight time are shown. Soon also attitude, heading and velocity will be added. Moreover it would be cool to have a config file on the sdcard to decide what to show on the overlay and where to place it on the screen.

Currently a parameter was added (OSD_TX_MODE) to choose between NTSC and PAL transmission mode (I still need to figure out if somehow it can be done automatically).

In the next days the remaining features will be pushed and also some RAM usage analysis will be done because in one situation I ran out of RAM.

### CODE REVIEW

**Comment by dagar** [comment]
If this is only running every 500ms you might as well grab all data you need every cycle. You can skip storing most of the state or even bothering to orb_check.

File: src/drivers/osd/osd.cpp [hunk 1]
```diff
@@ -0,0 +1,636 @@
...
+		ret |= add_battery_info(2, 1);
+	}
+
+	if (_local_position_valid) {
+		ret |= add_altitude_symbol(1, 3);
+		ret |= add_altitude(2, 3);
+	}
+
+	if (_arming_state > 0) {
+		float flight_time_sec = static_cast<float>((hrt_absolute_time() - _arming_timestamp) / (1.0e6));
```
Comment (dagar) [inline src/drivers/osd/osd.cpp]: Idea for later, this might be a useful thing to track more generally. Armed time, time in air, total distance flown, etc.

**Comment by DanielePettenuzzo** [comment]
Yes that makes sense. I still don't know at what frequency I will update. I will try to fly with it and then decide. Anyway it will be between 2 and 5 Hz.

**Comment by dagar** [comment]
> Currently the battery voltage, the accumulated current, altitude and flight time are shown. Soon also attitude, heading and velocity will be added. Moreover it would be cool to have a config file on the sdcard to decide what to show on the overlay and where to place it on the screen.

Depending on how much control you think people really need here I'd consider exposing it via parameters. It ends up being far more user accessible and to some extent self documenting. We could also have a handful of opinionated premade configurations selectable with a single parameter.

**Comment by LorenzMeier** [comment]
Yes, parameters are sufficient for now.

**Comment by dagar** [comment]
@DanielePettenuzzo worth rebasing and getting a partial version of this committed?

**Pull request merged**